src/utils.js

```diff
@@ -121,28 +121,34 @@ const ipRangeBlacklist = [
 ];
 
 /**
- * Check if a domain is blacklisted via IP ranges.
+ * Check if adresses are blacklisted via IP ranges.
  *
- * If it's not an IP already, resolve any addresses and check them all separately.
+ * @param {array} addresses             Adresses to check
+ * @returns {boolean}                   true if blacklisted
+ */
+function isBlacklisted(addresses) {
+    return Array.from(addresses).some(a => ipRangeCheck(a, ipRangeBlacklist));
+}
+
+/**
+ * Resolve a domain.
+ * 
+ * Throws an error if not resolvable.
  *
  * @param {string} domain           Domain to check
- * @returns {Promise<boolean>}      true if blacklisted
+ * @returns {array}                 The adresses resolved from the domain
+ * @throws                          On now resovlable domains
  */
-async function isDomainBlacklisted(domain) {
-    let addresses;
+ async function resolveDomain(domain) {
     if (!net.isIP(domain)) {
         try {
-            addresses = await dnsUtils.resolve(domain);
+            return await dnsUtils.resolve(domain);
```

Comment:
As far as I understood as the dnsUtils.resolve function is an async one, so we have to wait here for result otherwise returning a promise? However, I am not an JS expert at all.

Reply:
Yes that would be true if you were using the result of `dnsUtils.resolve` here directly, but it's been returned from this `async` function `resolveDomain`, so a promise will be returned anyway.

Reply:
@jaywink  It looks like there is a try/catch which should catch errors thrown by `dnsUtils.resolve` therefore the execution needs to be awaited.

Reply:
Good catch!

Actually, I now realize the behaviour is changing from the previous `isDomainBlacklisted` to never throwing to the combo of `isBlacklisted(await resolveDomain)` throwing if the domain cannot be resolved. This needs to be changed, otherwise we will have errors popping up the stack that were not doing so before (sorry, should have more comprehensive unit tests).

Reply:
Removed the new error -> Returns an empty array now.